jcstress-samples/src/main/java/org/openjdk/jcstress/samples/problems/classic/Classic_02_ProducerConsumerProblem.java

```diff
@@ -0,0 +1,325 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.openjdk.jcstress.samples.problems.classic;
+
+import org.openjdk.jcstress.annotations.*;
+import org.openjdk.jcstress.infra.results.III_Result;
+import org.openjdk.jcstress.infra.results.Z_Result;
+
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.ReentrantLock;
+
+import static org.openjdk.jcstress.annotations.Expect.ACCEPTABLE;
+import static org.openjdk.jcstress.annotations.Expect.ACCEPTABLE_INTERESTING;
+
+public class Classic_02_ProducerConsumerProblem {
+    /*
+        How to run this test:
+            $ java -jar jcstress-samples/target/jcstress.jar -t Classic_02_ProducerConsumerProblem
+    */
+
+    /*
+
+      ----------------------------------------------------------------------------------------------------------
+
+        This sample shows you how JCStress can help you to test solutions for the famous producer-consumer problem.
+        See https://en.wikipedia.org/wiki/Producer%E2%80%93consumer_problem for more information
+        about the problem and solutions.
+
+        The producer-consumer problem is about transferring items
+        from the producer(s) to the consumer(s) in a thread-safe way.
+        Some solutions support only one producer and one consumer
+        while other solutions don't mind many producers and many consumers.
```

Comment:
Formatting in this block should match the same block: right margin is too small here.